javascript/src/game.js

```diff
@@ -1,171 +1,173 @@
+  var popQuestions = new Array();
   var scienceQuestions = new Array();
-  var sportsQuestions  = new Array();
-  var rockQuestions    = new Array();
+  var sportsQuestions = new Array();
+  var rockQuestions = new Array();
 
-  var currentPlayer    = 0;
+  var currentPlayer = 0;
   var isGettingOutOfPenaltyBox = false;
 
-  var didPlayerWin = function(){
-    return !(purses[currentPlayer] == 6)
+  var didPlayerWin = () => {
+    return !(purses[currentPlayer] === 6);
   };
 
-  var currentCategory = function(){
-    if(places[currentPlayer] == 0)
-      return 'Pop';
-    if(places[currentPlayer] == 4)
-      return 'Pop';
-    if(places[currentPlayer] == 8)
-      return 'Pop';
-    if(places[currentPlayer] == 1)
-      return 'Science';
-    if(places[currentPlayer] == 5)
-      return 'Science';
-    if(places[currentPlayer] == 9)
-      return 'Science';
-    if(places[currentPlayer] == 2)
-      return 'Sports';
-    if(places[currentPlayer] == 6)
-      return 'Sports';
-    if(places[currentPlayer] == 10)
-      return 'Sports';
-    return 'Rock';
+  const categoryLookup = {
+    0: "Pop",
+    1: "Science",
+    2: "Sports",
+    3: "Rock",
+    4: "Pop",
+    5: "Science",
+    6: "Sports",
+    7: "Rock",
+    8: "Pop",
+    9: "Science",
+    10: "Sports",
+    11: "Rock"
   };
 
-  this.createRockQuestion = function(index){
-    return "Rock Question "+index;
+  var currentCategory = () => {
+    const playerPlace = places[currentPlayer];
+    return categoryLookup[playerPlace] || "Rock";
   };
 
-  for(var i = 0; i < 50; i++){
-    popQuestions.push("Pop Question "+i);
-    scienceQuestions.push("Science Question "+i);
-    sportsQuestions.push("Sports Question "+i);
-    rockQuestions.push(this.createRockQuestion(i));
-  };
+  this.createQuestion = (category, index) => `${category} Question ${index}`;
+  for (var i = 0; i < 50; i++) {
+    popQuestions.push(this.createQuestion("Pop", i));
+    scienceQuestions.push(this.createQuestion("Science", i));
+    sportsQuestions.push(this.createQuestion("Sports", i));
+    rockQuestions.push(this.createQuestion("Rock", i));
+  }
 
-  this.isPlayable = function(howManyPlayers){
-    return howManyPlayers >= 2;
-  };
-
-  this.add = function(playerName){
+  this.add = playerName => {
     players.push(playerName);
-    places[this.howManyPlayers() - 1] = 0;
-    purses[this.howManyPlayers() - 1] = 0;
-    inPenaltyBox[this.howManyPlayers() - 1] = false;
+    const placeIndex = this.howManyPlayers() - 1;
+    places[placeIndex] = 0;
+    purses[placeIndex] = 0;
+    inPenaltyBox[placeIndex] = false;
 
-    console.log(playerName + " was added");
-    console.log("They are player number " + players.length);
+    this.log(playerName + " was added");
+    this.log("They are player number " + players.length);
 
     return true;
   };
 
-  this.howManyPlayers = function(){
-    return players.length;
+  this.howManyPlayers = () => players.length;
+
+  var askQuestion = () => {
```

Comment:
We had debating on doing a bit of meta programming here and have something like:

```
this[`${categoryName()}Questions`].shift()
```

But decided that would add more headaches than the problem it solved (ie what happens when we text-search for all references to `popQuestions`?)